### PR TITLE
ZCS-13547: In-place upgrade migration implementation

### PR DESCRIPTION
**Pull request by vishnukantjangid**
Inplace upgrade implementation
-- Created two requests GetAllVolumesInplaceUpgradeRequest and ModifyVolumeInplaceUpgradeRequest.
-- These APIs are created for in-place upgrade implementation.

### CODE REVIEW

**Review by shrutig0510 — APPROVED** [review]

**Comment by dasiyogesh** [comment]
@vishnukantjangid these API implementation is done in modules porter in https://github.com/Zimbra/zm-modules-porter/pull/41 what are these APIs required for in mailbox?

**Comment by vishnukantjangid** [comment]
> @vishnukantjangid these API implementation is done in modules porter in [Zimbra/zm-modules-porter#41](https://github.com/Zimbra/zm-modules-porter/pull/41) what are these APIs required for in mailbox?

@dasiyogesh Only requests and responses are in modules porter for utilization but implementation is in zm-mailbox.

**Comment by dasiyogesh** [comment]
> 

Ok, usually its other way round, like for extension we declare the API's in mailbox and implement it in extension.

**Comment by vishnukantjangid** [comment]
> > 
> 
> Ok, usually its other way round, like for extension we declare the API's in mailbox and implement it in extension.

Ok. Here, I've used zm-modules-porter architecture.

**Review by rcyarrapothu — APPROVED** [review]

**Review by zimsuchitgupta — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java [hunk 1]
```diff
@@ -410,4 +429,179 @@ public static boolean isUnifiedVolume(int volumeId) {
...
+        }
+
+        StoreManager storeManager = StoreManager.getInstance();
+        if (storeManager.supports(StoreManager.StoreFeature.CUSTOM_STORE_API, String.valueOf(volInfo.getId()))) {
+            throw VolumeServiceException.INVALID_REQUEST("Operation unsupported, use zxsuite to edit this volume",
+                    VolumeServiceException.INVALID_REQUEST);
+        }
+
+        switch (vol.getStoreType()) {
+        case INTERNAL:
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java]: correct the indentation for switch cases

**Review by dasiyogesh — APPROVED** [review]

**Review by gautamdg — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java (hunk 1 above)
Comment (vishnukantjangid) [reply store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java]: done

**Review by shrutig0510 — APPROVED** [review]

**Pull request merged**